examples/brats18/pt/validators/monai_validator.py

```diff
@@ -0,0 +1,171 @@
+
+        self.train_config_filename = train_config_filename
+        self._validate_task_name = validate_task_name
+
+    def _initialize_validator(self, fl_ctx: FLContext):
+        # when the run starts, this is where the actual settings get initialized for trainer
+
+        # Epoch counter
+        self.epoch_of_start_time = 0
+
+        # Set the paths according to fl_ctx
+        self.app_root = fl_ctx.get_prop(FLContextKey.APP_ROOT)
+        fl_args = fl_ctx.get_prop(FLContextKey.ARGS)
+        self.client_id = fl_ctx.get_identity_name()
+        self.log_info(
+            fl_ctx,
+            f"Client {self.client_id} initialized at \n {self.app_root} \n with args: {fl_args}",
+        )
+
+        # Set the training-related contexts
+        train_config_file_path = os.path.join(self.app_root, self.train_config_filename)
+        self._validation_config(fl_ctx, train_config_file_path=train_config_file_path)
+
+    def _validation_config(self, fl_ctx: FLContext, train_config_file_path: str):
+        """monai traning configuration
+        Customized to invididual tasks
+        Needed for further training and validation:
+        self.model
+        self.device
+        self.transform_post
+        self.train_loader
+        self.valid_loader
+        self.inferer
+        self.valid_metric
+        """
+        pass
+
+    def _terminate_executor(self):
+        # collect threads, close files here
+        pass
+
+    def handle_event(self, event_type: str, fl_ctx: FLContext):
+        # the start and end of a run - only happen once
+        if event_type == EventType.START_RUN:
+            try:
+                self._initialize_validator(fl_ctx)
+            except BaseException as e:
+                error_msg = f"Exception in _initialize_validator: {e}"
+                self.log_exception(fl_ctx, error_msg)
+                self.system_panic(error_msg, fl_ctx)
+        elif event_type == EventType.END_RUN:
+            self._terminate_executor()
+
+    def local_valid(self, valid_loader, abort_signal: Signal):
+        self.model.eval()
+        with torch.no_grad():
+            correct, total = 0, 0
+            for i, (inputs, labels) in enumerate(valid_loader):
+                if abort_signal.triggered:
+                    return None
+                inputs, labels = inputs.to(self.device), labels.to(self.device)
+                outputs = self.model(inputs)
+                _, pred_label = torch.max(outputs.data, 1)
+
+                total += inputs.data.size()[0]
+                correct += (pred_label == labels.data).sum().item()
+            metric = correct / float(total)
+        return metric
+
+    def execute(
+        self,
+        task_name: str,
+        shareable: Shareable,
+        fl_ctx: FLContext,
+        abort_signal: Signal,
+    ) -> Shareable:
+        if task_name == self._validate_task_name:
+            # Check abort signal
+            if abort_signal.triggered:
```

Comment:
This check is unnecessary. The checking of abort_signal should be done in "long-running" logic.